### DISCUSSION TITLE
Support your GPUs for 8+ years, like Nvidia does, including gfx906 GPUs

### DISCUSSION THREAD

**Discussion by EwoutH** [ROCm/TheRock#8013] [General]
I was utterly amazed to read this in the ROCm 5.6 [release notes](https://rocm.docs.amd.com/en/latest/CHANGELOG.html#rocm-5-6-0):

> - AMD Instinct MI50, Radeon Pro VII, and Radeon VII products (collectively referred to as gfx906 GPUs) will be entering the maintenance mode starting Q3 2023. This will be aligned with ROCm 5.7 GA release date.
>   - No new features and performance optimizations will be supported for the gfx906 GPUs beyond ROCm 5.7
>   - Bug fixes / critical security patches will continue to be supported for the gfx906 GPUs till Q2 2024 (End of Maintenance [EOM])(will be aligned with the closest ROCm release)

The Vega 20 GPU (gfx906) is barely 5 years old. Even more, the Radeon VII, Radeon Pro VII and Instinct MI50 are still being sold!

If you want to know why everyone is going with Nvidia, there are a lot of them, but what they do at least well is they support they GPUs.

In the CUDA 12.x release from December 2022, Nvidia also [dropped support](https://docs.nvidia.com/cuda/cuda-toolkit-release-notes/index.html#cuda-libraries) for some GPUs:

> Support for the following compute capabilities is removed for all libraries:
> - sm_35 (Kepler)
> - sm_37 (Kepler)

Yes, you read that right, _Kepler_ GPUs. These are 2014 products. Not 2019 products like you're dropping.

So, hereby I would you strongly urge to not force us all to Nvidia again, by properly supporting your products.

**Comment by johnnynunez** [comment]
> I was utterly amazed to read this in the ROCm 5.6 [release notes](https://rocm.docs.amd.com/en/latest/CHANGELOG.html#rocm-5-6-0):
> 
> > * AMD Instinct MI50, Radeon Pro VII, and Radeon VII products (collectively referred to as gfx906 GPUs) will be entering the maintenance mode starting Q3 2023. This will be aligned with ROCm 5.7 GA release date.
> >   
> >   * No new features and performance optimizations will be supported for the gfx906 GPUs beyond ROCm 5.7
> >   * Bug fixes / critical security patches will continue to be supported for the gfx906 GPUs till Q2 2024 (End of Maintenance [EOM])(will be aligned with the closest ROCm release)
> 
> The Vega 20 GPU (gfx906) is barely 5 years old. Even more, the Radeon VII, Radeon Pro VII and Instinct MI50 are still being sold!
> 
> If you want to know why everyone is going with Nvidia, there are a lot of them, but what they do at least well is they support they GPUs.
> 
> In the CUDA 12.x release from December 2022, Nvidia also [dropped support](https://docs.nvidia.com/cuda/cuda-toolkit-release-notes/index.html#cuda-libraries) for some GPUs:
> 
> > Support for the following compute capabilities is removed for all libraries:
> > 
> > * sm_35 (Kepler)
> > * sm_37 (Kepler)
> 
> Yes, you read that right, _Kepler_ GPUs. These are 2014 products. Not 2019 products like you're dropping.
> 
> So, hereby I would you strongly urge to not force us all to Nvidia again, by properly supporting your products.

7900xtx and similars including laptops are working now, but not has official support. For the whole community it is already great news, they have finally realized that the whole opensource community (consumer gpus) and PhD students can greatly accelerate the progress of this library. Hopefully they will also give native windows support soon, although it is optional, almost everyone uses linux for AI.

**Comment by xcom169** [comment]
What about Vega56 / GFX9  is it still supported somehow?

**Comment by MelihDarcanxyz** [comment]
While the development of [NVK](https://gitlab.freedesktop.org/nouveau/mesa/-/tree/nvk/main) (a new open source Nvidia driver) is advancing, Nvidia is supporting their GPUs for at least 8 years, and already have the great (and mature) CUDA, I really can't see why a business or an individual should take a risk and go with AMD.

I want to support AMD, I really do. But the reasons above are hard to ignore.

Edit: It turns out NVK is only for Vulkan but it's an improvement still. We'll get there eventually with these.

**Comment by amayra** [comment]
as an RDNA user and I'm starting to be interested in AI there is no way to go with new AND GPU just because you don't went support old GPU like my RX 5700 XT 

CUDA looks so sexy from here when I'm trying experiment with Stable Diffusion  

I moved from windows and linux just to use hack environment variable "HSA_OVERRIDE_GFX_VERSION=10.3.0" to make Stable Diffusion work

**Comment by samuelpmish** [comment]
As a long time (10+ years) CUDA developer, I just bought a Radeon VII (because it was one of the few cards AMD appeared to actually support) so I could try out developing for AMD hardware, too. It's also a great card in the sense that it's one of the few options on the market (from any vendor) to have good FP64 performance at a reasonable price point.

It's disappointing to see AMD orphaning capable hardware like this so early in its lifetime.

**Comment by KEDI103** [comment]
As Radeon VII user and since 2005 using starting with Ati area bought always AMD but this thing force me why I am still going with AMD which I can't get support and suffer so much...
Nvidia users ridicule us which isn't nice as AMD user. 
AMD AI yt stream looks good but I can't get any benefit for AI but nvidia users crushing me. 
Even funny thing they removed Radeon vii page from AMD website. I was trying to compare my cards with modern cards I noticed one day puff it gone.
Should I regret my support and trust to AMD for years?
heres my problem right now...
https://github.com/RadeonOpenCompute/ROCm/issues/2314

**Comment by keryell** [comment]
I have bought recently 3 Radeon VII for some projects at work (which is... AMD). So this is sad to have them no longer supported soon. :-(
But this depends on what is really behind "entering the maintenance mode" or "Bug fixes will not be back ported to older ROCm releases" from https://rocm.docs.amd.com/en/latest/CHANGELOG.html#os-and-gpu-support-changes. At the end, if everything is *really* open-source in the sense that it can really be compiled everywhere with a unified robust build system and good open-source collaborative spirit at AMD, the open-source community could still do some improvement if it is required.
At the end it would not be that different from the competition where you cannot have anyway super modern features usable on latest GPU, except that if it is really open-source like ROCm, you can still extend the lifetime of some hardware beyond what marketing people care about.

**Comment by stalkerg** [comment]
Also, it seems like AMD team has some confusing intentions because AMDKFD interface in the Linux kernel still supports all cards started from GCN, and effectively only ROCm uses such an interface. (rusicls also thinking about it but not yet)
Because the kernel module/interface can exist only if has user space implementation this is a question for Linus and DRM maintainers.

I am a Vega56 user.

**Comment by stalkerg** [comment]
> At the end, if everything is really open-source in the sense that it can really be compiled everywhere with a unified robust build system and good open-source collaborative spirit at AMD, the open-source community could still do some improvement if it is required.

ROCm is open source but not open development. They do even not read the issue tracker usually.

**Comment by drtpotter** [comment]
Yes, please keep supporting gfx906 on ROCM for another few years at least! I use a Radeon VII for HIP training courses and it has by far the best double precision compute available in a consumer grade GPU. It is environmentally irresponsible to force users to discard hardware that still has plenty of life left!

**Comment by ev-dev** [comment]
+1 gfx906 support

I waited for **_years_** to  see ROCm come to Windows, only to find out that my Radeon VII, which was considered one of the best consumer GPUs only 2 years ago, wouldn't be included, not even in future releases! Please consider extending hardware support 🙏

**Comment by EwoutH** [comment]
> AMD Instinct MI50, Radeon Pro VII, and Radeon VII products (collectively gfx906 GPUs) enters maintenance mode in ROCm 6.0.

Nvidia still supports eight year old 2015 GPUs. AMD is dropping support for four year old 2019 GPUs.

When we finally thought we had a feasible Nvidia alternative, it was too good to be true.

Can’t believe we have to rely on Intel of all companies to offer a sustainable alternative.

**Comment by johnnynunez** [comment]
Because Nvidia not change cuda cores drastically over years. AMD is really new with cdna and rdna so in my opinion  they are working on these architectures right now, gcn is totally different.

the problem was the strategy of amd with raytracing and AI, not giving the sufficient importance. Now, nvidia is moving to path tracing and tensor cores and rt cores are quite fast to run algorithms and DL.

Rdna3 is not sufficiently strong in these areas but with rdna4 i’m pretty sure that they put AI hardware

**Reply by FlorianHeigl** [reply]
just doesn't hold up if you look at the marketing material for the Radeon VII Pro. There are jobs that it is supposed to handle as a product and those should be(come) available.

**Comment by johnnynunez** [comment]
Added support for additional GPU architectures.
Navi 3 series: gfx1100, gfx1101, and gfx1102.

**Comment by johnnynunez** [comment]
For me it's working very well.
Use my scripts and build tensorflow and pytorch with your id card.
https://github.com/johnnynunez/rocm_lab
https://github.com/ROCmSoftwarePlatform/tensorflow-upstream/pull/2327

**Comment by JustGitting** [comment]
Not sure if this should be added as another issue or placed here...

As pointed out by @stalkerg, rocm is "open source but not open development.". Where decisions to drop support for hardware is made behind closed doors with no input from normal users or clear reasons why.

It appears there is no technical reason why 5 year old AMD cards cannot be supported for a much longer period. I speculate that the reason for AMD's decision is data-centers and HPC customers don't use cards older then 4-5 years...

As others have pointed out, it is environmentally irresponsible to cut support or usability of a perfectly capable piece of kit.
It also denies many people around the world from learning computer science / ML / programming with AMD GPU's if they are forced to buy expensive hardware. This means students and those in poorer communities miss-out on the opportunities to learn in this area. Which leads people to go (back) to Nvidia as older and cheaper hardware is supported.

I, like most people, hope AMD do the right thing.

Community/AMD, is it possible to (re)add support for formally dropped cards?

From @stalkerg's [comment](https://github.com/ROCm/ROCm/issues/2308#issuecomment-1659518982) the limitation is not in the driver (amdgpu), but within rocm itself (?).

Is there any documentation or information about adding/maintaining support in rocm for unsupported GPUs?

Would AMD be open to accepting PR's from the community to support unofficial/dropped cards? 
Or could support be added in a modular way without direct inclusion in rocm?

**Comment by alann-sapone** [comment]
> as an RDNA user and I'm starting to be interested in AI there is no way to go with new AND GPU just because you don't went support old GPU like my RX 5700 XT
> 
> CUDA looks so sexy from here when I'm trying experiment with Stable Diffusion
> 
> I moved from windows and linux just to use hack environment variable "HSA_OVERRIDE_GFX_VERSION=10.3.0" to make Stable Diffusion work

Hi :)
I've been struggling to make it work and so far I didn't succeed...
Could you provide me any kind of guide (or a docker image) to finally be able to make my 5700XT work ? :)

Thanks =)

**Comment by PhilipDeegan** [comment]
I want to buy an Radeon VII pro, the specs are great and the price is great, but this is legitimately insane.

**Comment by SMH17** [comment]
> I was utterly amazed to read this in the ROCm 5.6 [release notes](https://rocm.docs.amd.com/en/latest/CHANGELOG.html#rocm-5-6-0):
> 
> > * AMD Instinct MI50, Radeon Pro VII, and Radeon VII products (collectively referred to as gfx906 GPUs) will be entering the maintenance mode starting Q3 2023. This will be aligned with ROCm 5.7 GA release date.
> >   
> >   * No new features and performance optimizations will be supported for the gfx906 GPUs beyond ROCm 5.7
> >   * Bug fixes / critical security patches will continue to be supported for the gfx906 GPUs till Q2 2024 (End of Maintenance [EOM])(will be aligned with the closest ROCm release)
> 
> The Vega 20 GPU (gfx906) is barely 5 years old. Even more, the Radeon VII, Radeon Pro VII and Instinct MI50 are still being sold!
> 
> If you want to know why everyone is going with Nvidia, there are a lot of them, but what they do at least well is they support they GPUs.
> 
> In the CUDA 12.x release from December 2022, Nvidia also [dropped support](https://docs.nvidia.com/cuda/cuda-toolkit-release-notes/index.html#cuda-libraries) for some GPUs:
> 
> > Support for the following compute capabilities is removed for all libraries:
> > 
> > * sm_35 (Kepler)
> > * sm_37 (Kepler)
> 
> Yes, you read that right, _Kepler_ GPUs. These are 2014 products. Not 2019 products like you're dropping.
> 
> So, hereby I would you strongly urge to not force us all to Nvidia again, by properly supporting your products.

I totally agree, but I have to correct you about the age of the no more supported Kepler GPUs mentioned for comparison:  actually they are **2012** GPUs!

**Reply by cb88** [reply]
This is insane, the REASON Nvidia has so much mind share... is you CAN buy old GPUs and learn on them, and that translates directly into writing CUDA code for HPC.... do you expect college students to buy MI350X next year instead of writing code for MI60 which they can afford easily?

**Reply by SMH17** [reply]
Largest is the number of hardware that support an API, greatest is the incentive to include the support for that API by third parties’ developers and industry. Obviously if an API works only with a small combination of hardware and software, there is less interest to invest resources and time to support a niche. Most developers and companies will not go to buy new AMD Instinct GPUs just for the sake to work on ROCm, that is less popular and mature than CUDA, with the fear that such hardware may be also become unsupported in few years as happened to AMD Instinct MI25 based on GCN 5.0, it is AMD that have to make it as accessible as possible to maximize people involvement, confidence on long term support and push its diffusion. To boost its adoption, they should have ROCm support extended to all Radeon included the ones of the GCN generation and APUs that are massively popular and with proper ROCm support would have even allowed to experiment AMD technology with LLM on very low budget using system memory in UMA mode; and despite performance level cannot bee optimal also on older and\or weaker GPUs, still the gargantuan user basin gained would benefit the whole AMD ecosystem.

**Comment by stalkerg** [comment]
@ppanchad-amd do you have any valid comment here? Is it a official answer?

**Comment by j-jasz** [comment]
Reposting an opinion from Phoronix that quite well summarizes my stance on hardware support on ROCm. 
https://www.phoronix.com/forums/forum/linux-graphics-x-org-drivers/open-source-amd-linux/1508304-amd-talks-up-imminent-rocm-6-3-with-big-performance-gains-new-features?p=1508586#post1508586

> This is f***ing shameful AMD. Card support is the biggest thing that separates you from CUDA. Card support is the one big reason devs don't want to go fucking near your stack unless bean counters come down from on high and force it as the "bang for buck" option. Card support. Card support. Card support.

> Nvidia supports SEVERAL GENERATIONS of cards with CUDA, across both enterprise and consumer cards. Devs prototype apps with consumer cards, deploy on enterprise cards. It's not rocket science. If your stack doesn't run on the cards people have, they won't run it.

> You need to support ROCm, *officially* and with no compromises, on every market-relevant consumer card and enterprise card. It would be absurd for a game studio to say "sorry, we're dropping support for our game on the nvidia 30XX series now that 50XX is coming out" when there are even lots of 20XX cards around. Furthermore, this asinine bullshit of "well maybe this card works, the code is still there for it but it's not official" has to end. Today. The only reason I think AMD can't bring themselves to draw a hard line in the sand and drop code for cards they don't officially support is it would reveal just how vanishingly small their support pool actually is.

>As far as I'm concerned. ROCm is dead on arrival to me. I will never, ever use it, no matter how much you improve performance, no matter how many features you add, unless you address this massive gulf of support. The ROCm compute stack does not exist. I have to assume that no matter what card I have, it is either not supported or will be broken by a change tomorrow. I patiently wait for MESA to do your job for you and get rusticl together, providing the reliable, works-on-every-card experience you do not comprehend the importance of.

>PS. I'm sorry you did something stupid and didn't implement an equivalent of nvidia's PTX (or any standard intermediate representation) and instead required apps to compile down to specific cards, if that is indeed a limiting factor. Maybe you should think about implementing that if you want to ease your own maintenance burden and the burden of people attempting to support you. 

Unless AMD addresses those issues seriously I'd rather root for any other alternative that be NVIDIA opensourcing their driver stack, Intel pushing oneAPI, Vulkan compute or anything else that can give us working compute stack on Linux. Even thou I have AMD GPU in my own Linux machine, when it came to advising purchases for an institution that I work for I advocated for NVIDIA specifically for the reason explained above.

**Reply by johnnynunez** [reply]
> Reposting an opinion from Phoronix that quite well summarizes my stance on hardware support on ROCm. https://www.phoronix.com/forums/forum/linux-graphics-x-org-drivers/open-source-amd-linux/1508304-amd-talks-up-imminent-rocm-6-3-with-big-performance-gains-new-features?p=1508586#post1508586
> 
> > This is f***ing shameful AMD. Card support is the biggest thing that separates you from CUDA. Card support is the one big reason devs don't want to go fucking near your stack unless bean counters come down from on high and force it as the "bang for buck" option. Card support. Card support. Card support.
> 
> > Nvidia supports SEVERAL GENERATIONS of cards with CUDA, across both enterprise and consumer cards. Devs prototype apps with consumer cards, deploy on enterprise cards. It's not rocket science. If your stack doesn't run on the cards people have, they won't run it.
> 
> > You need to support ROCm, _officially_ and with no compromises, on every market-relevant consumer card and enterprise card. It would be absurd for a game studio to say "sorry, we're dropping support for our game on the nvidia 30XX series now that 50XX is coming out" when there are even lots of 20XX cards around. Furthermore, this asinine bullshit of "well maybe this card works, the code is still there for it but it's not official" has to end. Today. The only reason I think AMD can't bring themselves to draw a hard line in the sand and drop code for cards they don't officially support is it would reveal just how vanishingly small their support pool actually is.
> 
> > As far as I'm concerned. ROCm is dead on arrival to me. I will never, ever use it, no matter how much you improve performance, no matter how many features you add, unless you address this massive gulf of support. The ROCm compute stack does not exist. I have to assume that no matter what card I have, it is either not supported or will be broken by a change tomorrow. I patiently wait for MESA to do your job for you and get rusticl together, providing the reliable, works-on-every-card experience you do not comprehend the importance of.
> 
> > PS. I'm sorry you did something stupid and didn't implement an equivalent of nvidia's PTX (or any standard intermediate representation) and instead required apps to compile down to specific cards, if that is indeed a limiting factor. Maybe you should think about implementing that if you want to ease your own maintenance burden and the burden of people attempting to support you.
> 
> Unless AMD addresses those issues seriously I'd rather root for any other alternative that be NVIDIA opensourcing their driver stack, Intel pushing oneAPI, Vulkan compute or anything else that can give us working compute stack on Linux. Even thou I have AMD GPU in my own Linux machine, when it came to advising purchases for an institution that I work for I advocated for NVIDIA specifically for the reason explained above.

RDNA4 is on ROCM 6.3. AMD starts with 4 gpus rdna3 support, 7900XT, 7900XTX, 7800 AI, 7900 AI. 
They are done big steps in last year, and they are working parity between linux and windows, and support more distros.
Rocm6.3 is coming but rocm7 comes with unified stack.

RDNA4 is coming with sistolic arrays, so it will be a big improve for inference. RDNA5 will be UDNA
https://www.phoronix.com/news/AMD-ROCm-6.3-Features

**Reply by cb88** [reply]
A hodge podge of releases that "support" the range of GPUs.... is very cumbersome. AMD needs to start being about we make things work NOW, rather than wait till next release. All the GPUS of the last decade need to work now under one roof.

**Reply by johnnynunez** [reply]
> A hodge podge of releases that "support" the range of GPUs.... is very cumbersome. AMD needs to start being about we make things work NOW, rather than wait till next release. All the GPUS of the last decade need to work now under one roof.

I understand your part, but I think the previous ranges have very poor performance, even rdna3 in computation and in inference is not too good. Thus, the change of philosophy with UDNA. I have w7900 dual slot

**Reply by j-jasz** [reply]
@johnnynunez please refrain from quoting the entire post of someone you just want to respond in a general way!

> RDNA4 is on ROCM 6.3.

Have you at least read the message I quoted and what I have written myself? Or you are responding just for the sake of having a conversation? You are trying to make a point about future generations, I'm talking about supporting past ones.

> but I think the previous ranges have very poor performance, even rdna3 in computation and in inference is not too good.

Tunnel vision response. COMPUTE =! AI. Let me repeat this: **COMPUTE =! AI**

You clearly do not take into account that ROCm can and **IS** used by software other than AI. Case in point - Blender (and hopefully many other in the future). I and many others do not care that AMD 3D rendering performance (in applications like Blender) is on the lower side as long the GPUs are cheaper and have stable driver stack. Radeons usually have a lot of VRAM and that is very useful for rendering bigger scenes even if the rendering is slower vs NVIDIA. Driver stability and platform longevity are much more important than absolute performance. This is the case for both 3D rendering market segment and ROCm developers, as it was stated in the post I quoted.
For the record, aside from using Blender I also use ROCm for AI inference. And I will even argue for the platform stability and longevity for DAW, CAD, medical, etc, folks as well even I'm not using their software myself!

Sorry for the harsh tone. It's not personal. I have very low tolerance for meandering discussion style and ignoring the fundamentals for the sake of the hype.

**Comment by Snektron** [comment]
There seems to be some effort going on about supporting SPIR-V based kernels of ROCm: https://github.com/llvm/llvm-project/pull/89796. Speaking from experience developing ROCm projects, these things usually take quite some time, and I expect (or rather hope) that this SPIR-V supports lands in the rest of the ROCm stack sooner or later. This would enable ROCm projects to support a wide array of different GPUs more easily. This the equivalent of PTX that the phoronix comment points out.

On a side note, HSA (Linux ROCm's backing driver interface) already supported a PTX equivalent in the past, HSAIL. I guess ROCm never adopted it...

**Reply by stalkerg** [reply]
as I understand, the IR is not the main issue here.

**Reply by Snektron** [reply]
As you noted before, "unsupported" GPUs work mostly fine with KFD, this maybe just needs some testing infrastructure and a label to say that it actually really works. From the POV of ROCm projects like the ones we (where I work) maintain, compiling for all those different GPUs takes ages and is a real pain. Each extra architecture (each `gfx<number>`) increases compile times linearly. Have you checked how big rocBLAS is already? IR is absolutely the main issue here.

That reminds me, "generic" targets for each generation recently started appearing in LLVM: https://llvm.org/docs/AMDGPUUsage.html#amdgpu-generic-processor-table. I always wondered what the point of splitting each die into a separate architecture was, considering that they all have the same instructions anyway (this is also why `HSA_OVERRIDE_GFX_VERSION` works so well). Seems that this would finally fix that.

Either way it looks like they are aware of the problem and are working on it. I'm not trying to justify deprecating MI50 though, and as long as we don't find a better GPU to replace it with in our CI, it will still be tested for the projects we maintain.

**Reply by j-jasz** [reply]
That is good to hear actually.
Just out of curiosity is IR relatively stable and different ROCm versions compile to IR in a way that it can be run on any GPU that can process that IR? Is this what NVIDIA did with forcing OptiX to run on CUDA cores by compiling to PTX that can be understood by a GPUs that do not have Tensor cores (like 1080ti)?

**Reply by Snektron** [reply]
That would work with the SPIR-V based approach, yes. The "generic" targets are still machine code like before, and so they wouldn't. I guess its just a subset of the instructions that each GPU in the generation supports, if there are any differences...

**Reply by LunNova** [reply]
Requested an intermediate bytecode in https://github.com/ROCm/ROCm/issues/3985 and it was closed as not planned.

**Reply by LunNova** [reply]
https://github.com/ROCm/ROCm/issues/3985#issuecomment-2546161823

> Sorry about that, you are correct, there is work in progress to support compiling down to an AMDGCN flavoured SPIR-V. However, this will be a secondary option alongside the existing approach to compile down to ISA (which provides better optimization and lower overhead at runtime as mentioned above) rather than a replacement, so users will be able to choose between compiling to ISA or to SPIR-V.

**Reply by MathiasMagnus** [reply]
Hi @Snektron, came across this discussion by accident.

> I guess its just a subset of the instructions that each GPU in the generation supports, if there are any differences..

You guessed right. Pop up https://llvm.org/docs/AMDGPUUsage.html and Ctrl+F "Target Restrictions" which will show you what features are lost on each of the `gfxY-Z-generic` targets. I too thought it's nuances only, but they are mostly pretty serious: losing VGPRs, losing important instructions (SALU floating ops, one of the selling points of the generation)... at this point I'm not entirely sure these generic targets make any sense. I too hope that the SPIR-V stuff land sooner, rather than later. I have both the [comgr](https://github.com/ROCm/llvm-project/tree/amd-staging/amd/comgr) folder and [`clr/rocclr/device/device.hpp`](https://github.com/ROCm/clr/blob/amd-staging/rocclr/device/device.hpp#L205) bookmarked to monitor SPIR-V progress and the hopeful landing of `cl_khr_il_program`.

It can't come soon enough. Maybe add [`VK_AMD_hip_kernel_launch`](https://registry.khronos.org/vulkan/specs/latest/man/html/VK_NV_cuda_kernel_launch.html) while they're at it. (Or just expose the NV variant, although that's a grossly underdocumented feature. It's also absent from vk.xml, so it's hard to meaningfully extend its API with AMD-specific structure types.)

**Comment by KEDI103** [comment]
https://github.com/pytorch/pytorch/pull/142827

**Reply by v1ckxy** [reply]
This people is just plain nuts. Quite a good example of enshittification.

Those gfx900 are RX Vega 64 from 2017-20 (GCN 5.0) and the gfx906, MI50/MI60 from 2018-2022 (GCN 5.1) 

|  _reduce the increasing size of ROCm binaries._

Who in the earth will complain about that, lol🤨?

**Comment by JustGitting** [comment]
ROCm Device Support Wishlist ROCm/legacy-rocm-build#4276
https://github.com/ROCm/ROCm/discussions/4276

**Comment by v1ckxy** [comment]
I just saw this thread and to put it simple...
What is this people thinking?
**We are in 2025 and we are still the same or worse.**

The entire mid-range and low-end unsupported (laptops with 680m/780m integrated, better not talk about those with VEGA graphics) and drivers that break the performance of ROCm.

**Comment by FlorianHeigl** [comment]
I just saw... 
HSA_OVERRIDE_GFX_VERSION isn't mentioned on https://rocm.docs.amd.com/projects/HIP/en/docs-develop/reference/env_variables.html
Please, people at AMD - almost any user will need this at some point. Some will need it always. And it's not even mentioned on the main page where it ought to be.
This mental disconnect needs some glue, start right at the authoritative docs and maybe it will hurt a bit but it will hurt the users much less afterwards, and it will make it easier to start solving the problem. (I mean the in progress one of turning the ecosystem around, not the problem the users might run into because of it. Since they'll still need to set that variable, in the better case, or have a non-useful product in the worse. But the majority will be helped by the flag)
Oh and, less funny unless you're equiped with some gallow humor, the ROCm dev portal doesn't have download links. Only pretty icons. Maybe do something about that too? ;-)

**Reply by cb88** [reply]
It would be even better if it *just worked* ... as in generated whatever the optimal code that could run on a given GPU by default instead of disabling things by default and making things hard for no reason. Acutally that's what every other API/SDK out there I have ever seen does....

**Reply by IMbackK** [reply]
gfxX-generic finally fixes this, at lest when everything finally switches over to that...

**Comment by FlorianHeigl** [comment]
Shoutout to why re-instating MI50 support would simply be the right thing to do:
https://www.youtube.com/watch?v=EnzYYKhXXV0
community is making it work now because the demand and purpose exist, so AMD should make it frictionless, i.e. stop removing old files as mention in there after 10:40. There's no benefit for anyone by these restrictions that don't line up with requests. just waste of human lifetime.

**Comment by GZGavinZhao** [comment]
> `gfxX-generic` finally fixes this, at lest when everything finally switches over to that...

ROCm/rocm-libraries#3549 is hopefully one of the first steps towards this effort. If this approach can be replicated, it theoretically enables us to support every GPU since 2017 (every `gfx9XX` and `gfx1XXX` GPU).

The rocBLAS library fixed in this PR is the main blocker for bringing such support to llama.cpp. PyTorch needs a bit more work just due to the extra number of libraries it depends on.

**Reply by GZGavinZhao** [reply]
Oops I forgot Baffin and Polaris 31 are (I think) the last two GCN GFX8 GPU released after the debut of GFX9 RX Vega 64 GPUs in 2017, but my general point still holds.

**Comment by erkinalp** [comment]
if anyone knows why, it's @saadrahim

**Comment by jhstatewide** [comment]
This thread is the single biggest advertisement for NVidia / CUDA. AMD likes to talk and say "ohhh I wish people bought our cards, we want to be leaders in machine learning". But, then when it comes time to do the work.... little/no support.

ROCm is a second fiddle. My next build will be CUDA based, hands down. Sorry AMD, the hardware can be great but it's trash without decent, extended hardware support.

**Reply by KEDI103** [reply]
> This thread is the single biggest advertisement for NVidia / CUDA. AMD likes to talk and say "ohhh I wish people bought our cards, we want to be leaders in machine learning". But, then when it comes time to do the work.... little/no support.
> 
> ROCm is a second fiddle. My next build will be CUDA based, hands down. Sorry AMD, the hardware can be great but it's trash without decent, extended hardware support.



Yep after this painful months

https://github.com/pytorch/pytorch/issues/103973

 I got so enought as 2005 AMD/ATI user Now I got astral RTX 5090 LC OC rocking with anything on AI. And its really awesome not like ROCM hell. Because If you got amd basicly you can't do AI needs cuda things.

No cuda nearly mean no AI and amd rocm support so bad and killing card so early is just enought to not even touch them.

Bonus content: If you got AMD card AMD sofware lock on windows server make sure you will not use their card by blocking instalation past one not have it but last one have it. That bonus reason never gona buy AMD gpu again never and never gona suggest even I change people mind don't even touch AMD gpu they are killing support or useless against CUDA or sofware block.

 Anti consumer things not cool you know AMD.
 
 I still bought AMD ryzen 9950x3d  cpu but if intel do something I reconsider on my next build too after I see this threatment as consumer at amd gpu deparmant...

**Reply by stalkerg** [reply]
> No cuda nearly mean no AI and amd rocm support so bad and killing card so early is just enought to not even touch them.

it's absolutely wrong statement, you just too far from real AI or LLM development.

**Reply by KEDI103** [reply]
> wrong statement, you just too far from real AI or LLM development.

I don't only care LLM I said because  I need AI mocap on windows which depends on nvidia also AMD not even in windows pythorch.
+ cumesh + nvdiffrast + nvdiffrec only work on nvidia cards because of nvidia technology lots of thing depends on CUDA not ROCM cause you can't use it. So my state very clearly and correct its not even need argue. If you got AMD you gots problems or you can't just can't on some AI workloads depends CUDA or nvidia sdk.

So basily some AI worloads only work with CUDA or nvidia sdk you No cuda nearly mean no AI still is going.
I am sure I can extend my list.

**Reply by cb88** [reply]
> > wrong statement, you just too far from real AI or LLM development.
> 
> I don't only care LLM I said because I need AI mocap on windows which depends on nvidia also AMD not even in windows pythorch.
> 
>     * cumesh + nvdiffrast + nvdiffrec only work on nvidia cards because of nvidia technology lots of thing depends on CUDA not ROCM cause you can't use it. So my state very clearly and correct its not even need argue. If you got AMD you gots problems or you can't just can't on some AI workloads depends CUDA or nvidia sdk.
> 
> 
> So basily some AI worloads only work with CUDA or nvidia sdk you No cuda nearly mean no AI still is going. I am sure I can extend my list.

AMD acutally has pytorch on windows now for newer GPUs at least.

**Comment by Maxritz** [comment]
[ROCm_7x_gfx1031_Hard_Bridge_v2.md](https://github.com/user-attachments/files/25383607/ROCm_7x_gfx1031_Hard_Bridge_v2.md)

if someone wants to use this as a reference

**Comment by edhemphill** [comment]
June 2026, has this situation improved at all? I can't find much evidence it has. Looking to purchase some used gear.

**Reply by Maxritz** [reply]
seems like.. 1031 can sort of work a bit on 7.13 but not with LMstudio for sure
